lib/domain.js

```diff
@@ -46,8 +46,11 @@ const { WeakReference } = internalBinding('util');
 
 // Overwrite process.domain with a getter/setter that will allow for more
 // effective optimizations
+// Never directly access _domain, use the getter to ensure a consistent hook
+// for user code
 const _domain = [null];
 ObjectDefineProperty(process, 'domain', {
+  configurable: true,
```

Comment:
What about the `domain` property [defined on error objects thrown when a domain is on the domains' stack](https://github.com/nodejs/node/blob/ec8676c6e29c0b8855044314f8740370277483f9/lib/domain.js#L217-L222), wouldn't that still be set with the changes on this PR, and still leak information from SES' perspective?

Reply:
`_errorHandler` is only ever called by core through an access of `process.domain` so this should work to my knowledge I can ask on thursday for extra clarity on if they have seen problems with the property on errors.

Reply:
Ah that's correct, but then in this case the current domains' implementation doesn't handle `process.domain` not evaluating to an object with a callable `._errorHandler` property (it exits with a `TypeError`). Should the implementation of `lib/domain.js` in this PR be more defensive against those use cases?

Reply:
I am unclear and can ask again on thursday but it seems i could make it more robust regardless and thats a good idea.

Reply:
> I am unclear and can ask again on thursday

Just out of curiosity, what is happening on Thursday that is relevant to this PR?

Reply:
the ocap JS people meet to discuss SES, realms, etc. every thursday at the SES strategy meeting (can be seen on the [TC39 calendar](https://calendar.google.com/calendar/embed?src=mozilla.com_l7b50itpaa9bnrvr61nebqrne8%40group.calendar.google.com&ctz=America%2FChicago))

Reply:
@bmeck Ah cool! If it'd help for me to join the meeting just to provide feedback on impact on the domains' use cases, let me know, I'd be happy to join.

Reply:
it would be of great help if you could attend one week

Reply:
Sounds good, let's sync on IRC to figure out when I might be able to help in those meetings 👍